### PR TITLE
Adding -j parameter to the make.

### PR DESCRIPTION
**Pull request by fatihbm**
This parameter optional but useful if it has more than one cpu.

### Description

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by tokers** [comment]
@fatihbm Could you rename the PR title according to https://github.com/zeke/semantic-pull-requests.

File: docs/en/latest/how-to-build.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ Follow the steps below to install Apache APISIX via the source release package.
   # Switch to the apisix-${APISIX_VERSION} directory
   cd apisix-${APISIX_VERSION}
   # Create dependencies
-  make deps
+  make deps # -j$(nproc) ## use this parameter for make building go fast.
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:127]: The `make deps` will call luarocks to download dependencies. Even the make is running in parallel, luarocks can't download in parallel.

**Comment by leslie-tsang** [comment]
Hello there, I afraid this feat can't do help for more than one CPU scenario.

`make deps` will use luarocks to install deps defined in `rockspec/apisix-master-0.rockspec`. which is not parallelizable. :)

**Comment by tzssangglass** [comment]
as @spacewander said, this does not help with parallel downloads (it is not actually a make compilation process).

I will turn it off because there is no practical use for it.

**Pull request closed without merging**